azerothcore/azerothcore-wotlk
# fix(Core\Scripts):Venomhide Hatchling multiple fixes

## Changes Proposed:
- Venomhide Hatchling should offer quests only to summoner
- Venomhide Hatchling should emote when you throw food at him
- Venomhide Hatchling should have his model based on how many baby tooths you have

## Issues Addressed:

- 1.1 Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8779
- 1.2 Closes https://github.com/chromiecraft/chromiecraft/issues/1802
- 2.1 Closes https://github.com/chromiecraft/chromiecraft/issues/1965
- 2.2 Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8409
- 2.3 Closes https://github.com/azerothcore/azerothcore-wotlk/pull/8556
- 3.1 Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8107
- 3.2 Closes https://github.com/chromiecraft/chromiecraft/issues/1871
- 3.3 Closes https://github.com/azerothcore/azerothcore-wotlk/pull/8557
- 
## SOURCE:


## Tests Performed:

- 1.1 and 1.2 > get 2 chars, one with quest (13906) and other without, summon npc hatchling and notice you can take quests from him without having the main quest or being the owner; also you cannot take quests from another character's hatchling even if you have the main quest 
- 2.1 and 2.2 > .quest add 13906 and summon the pet, regarding how many Baby Tooths you have, he will have the same model
- 3.1 and 3.2 Throw some food at him, for example Silithid Meat, no emote at all


## How to Test the Changes:


1. 1.1 and 1.2 > get 2 chars, one with quest (13906) and one without, summon hatchling and notice only the summoner can take quests from him, even if the 2nd char has the main quest or not
2. 2.1. and 2.2 .quest add 13906 and summon pet, he has various models based on how many Baby Tooths you have in the bag, the more Tooths, the bigger the pet is, ex:
( X < 6 ) = Baby
( X >= 6 and X < 11 ) = Bigger baby with reptile eyes
( X >= 11 and X < 16 ) = Adolescent raptor
( x >= 16 ) = Full-grown raptor
3. 3.1 and 3.2 Throw some food at him, he will emote that he is eating.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
